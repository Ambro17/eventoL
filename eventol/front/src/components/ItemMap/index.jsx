import React from 'react';
import PropTypes from 'prop-types'
import './index.scss';


export default class ItemMap extends React.Component {
  propTypes = {
    place: PropTypes.string,
    title: PropTypes.string,
    url: PropTypes.string,
    attendees: PropTypes.string,
    overview: PropTypes.string,
    uid: PropTypes.string,
    sliderId: PropTypes.string
  };

  getMapId(){
    const {uid, sliderId} = this.props;
    return `${sliderId}${uid}`;
  }

  loadMap(){
    const {place} = this.props;
    const {geometry} = JSON.parse(place);
    const event_lat = geometry.location.lat;
    const event_lon = geometry.location.lng;
    const mapId = this.getMapId();
    const map = L.map(mapId, {
        scrollWheelZoom: false,
        zoomControl    : false,
        dragging       : false,
        keyboard       : false,
        doubleClickZoom: false,
        touchZoom      : false,
        boxZoom        : false,
        minZoom: 11,
        maxZoom: 11
    }).setView([event_lat, event_lon], 5);
    L.tileLayer('https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png', {
        attribution: ''
    }).addTo(map);
    map.attributionControl.setPrefix('');
    L.marker([event_lat, event_lon]).addTo(map);
  }

  componentDidMount(){
    const uid = this.getMapId();
    try {
      this.loadMap();
    } catch(e){}
    document.getElementById(uid).classList.remove('max-size')
  }

  render(){
    const {title, url, attendees, overview, uid, sliderId} = this.props;
    const mapId = this.getMapId();
    return (
      <div id={mapId} className="Item max-size">
        <a href={url}>
          <div className="overlay">
            <div className="title">{title}</div>
            <div className="rating">{gettext('Attendees')}: {attendees}</div>
            <div className="plot">{overview}</div>
          </div>
        </a>
      </div>
    )
  }
}
