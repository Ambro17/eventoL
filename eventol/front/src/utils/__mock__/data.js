export const event1 = {
  attendees: 10,
  backdrop: 'event1backdrop',
  eventSlug: 'event1slug',
  overview: 'event1overview',
  place: 'event1place',
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 1',
  url: 'event1Url',
  id: 1,
};

export const event2 = {
  attendees: 25,
  backdrop: 'event2backdrop',
  eventSlug: 'event2slug',
  overview: 'event2overview',
  place: 'event2place',
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 2',
  url: 'event2Url',
  id: 2,
};

export default {
  event1,
  event2,
};
