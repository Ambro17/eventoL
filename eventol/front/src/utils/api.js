import Cookies from 'js-cookie';


export function getUrl(url){
  return fetch(url, {
    method: 'GET',
    credentials: 'same-origin',
    headers: {
        'X-CSRFToken': Cookies.get('csrftoken'),
        'Accept': 'application/json',
        'Content-Type': 'application/json'
     }
   }).then(resp => resp.json())
}
