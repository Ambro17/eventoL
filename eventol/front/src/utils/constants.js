export const SLIDER_BASE_SETTINGS = {
  variableWidth: false,
  dots: true,
  infinite: false,
  speed: 500,
  slidesToShow: 4,
  slidesToScroll: 4,
  initialSlide: 0,
  responsive: [{
    breakpoint: 1290,
    settings: {
      slidesToShow: 3,
      slidesToScroll: 3,
      infinite: true,
      dots: true
    }
  }, {
    breakpoint: 900,
    settings: {
      slidesToShow: 2,
      slidesToScroll: 2,
      initialSlide: 2
    }
  }, {
    breakpoint: 610,
    settings: {
      slidesToShow: 1,
      slidesToScroll: 1
    }
  }]
};

export const HOME_REQUIRED_FIELDS = 'event_slug,place,image,name,attendees_count,abstract';
export const REPORT_REQUIRED_FIELDS = 'name,event_slug,email,location,report,id';
