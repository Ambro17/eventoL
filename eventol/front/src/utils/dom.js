const focusOn = id => {
  document.getElementById(id).focus();
};

const DOM = {focusOn};
export default DOM;
