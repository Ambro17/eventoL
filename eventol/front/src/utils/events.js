/* eslint-disable-next-line import/prefer-default-export */
export const getEvent = (pk, fields) => ({pk, fields});
