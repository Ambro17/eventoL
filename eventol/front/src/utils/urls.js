export const API_URL = '/events/api/';
export const EVENTS_API_URL = `${API_URL}events/`;

export const WS_URL = `ws://${window.location.host}/`;
export const EVENTS_WS_URL = `${WS_URL}updates/events/`;

export const getTagUrl = slug => `/tags/${slug}`;

export const goToUrl = url => {
  window.location.href = url;
};
