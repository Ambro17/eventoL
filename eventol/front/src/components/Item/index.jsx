import React from 'react';
import ListToggle from '../ListToggle';
import ItemMap from '../ItemMap';
import './index.scss';


export default class Item extends React.Component {

  render(){
    const {title, url, attendees, overview, backdrop, slug} = this.props;
    if (!backdrop) {
      return <ItemMap {...this.props} />;
    }
    return (
      <div className="Item" style={{backgroundImage: 'url(' + backdrop + ')'}} >
        <a href={url}>
          <div className="overlay">
            <div className="title">{title}</div>
            <div className="rating">Asistentes: {attendees}</div>
            <div className="plot">{overview}</div>
          </div>
        </a>
      </div>
    );
  }
};
