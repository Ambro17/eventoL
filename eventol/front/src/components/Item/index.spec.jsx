import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../ItemMap', () => 'ItemMap');

jest.mock('../../utils/urls', () => ({
  getTagUrl: jest.fn().mockImplementation(() => 'tag-url'),
  goToUrl: jest.fn(),
}));

import ItemMap from '../ItemMap';

import {event1} from '../../utils/__mock__/data';
import {getTagUrl, goToUrl} from '../../utils/urls';

import Item from '.';

describe('Item', () => {
  let component;
  let tree;
  let data;
  let sliderId;

  const renderItem = () => {
    component = renderer.create(<Item data={data} sliderId={sliderId} />);
    tree = component.toJSON();
  };

  beforeEach(() => {
    data = undefined;
    sliderId = undefined;
  });

  test('Default render', () => {
    renderItem();
    expect(tree).toMatchSnapshot();
  });

  test('should shows ItemMap when data has not backdrop', () => {
    data = {backdrop: undefined};
    renderItem();
    expect(tree).toMatchSnapshot();

    const itemMap = component.root.findByType(ItemMap);
    expect(itemMap).toBeDefined();

    expect(itemMap.props.sliderId).toEqual(sliderId);
    const props = [
      'attendees',
      'eventSlug',
      'overview',
      'place',
      'title',
      'url',
    ];
    props.forEach(prop => expect(itemMap.props[prop]).toEqual(data[prop]));
  });

  test('Render event 1', () => {
    data = event1;
    sliderId = event1.id;
    renderItem();
    expect(tree).toMatchSnapshot();
  });
});
