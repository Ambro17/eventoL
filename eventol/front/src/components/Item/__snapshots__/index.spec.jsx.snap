// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Default render 1`] = `
<ItemMap
  attendees={null}
  url={null}
/>
`;

exports[`Item Render event 1 1`] = `
<div
  className="item"
  style={
    Object {
      "backgroundImage": "url(event1backdrop)",
    }
  }
>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="overlay"
    >
      <div
        className="title"
      >
        event 1
      </div>
      <div
        className="rating"
      >
        Attendees: 10
      </div>
      <div
        className="rating tags"
      >
        Tags
        : 
        <a
          href="tag-url"
        >
          slug-name
        </a>
      </div>
      <div
        className="plot"
      >
        event1overview
      </div>
    </div>
  </div>
</div>
`;

exports[`Item should shows ItemMap when data has not backdrop 1`] = `<ItemMap />`;
